Withdraw feature

sag333ar opened the pull request.

sag333ar committed:
work in progress for withdraw

miloridenour commented on src/lib/sendswap/stages/withdraw/LightningWithdraw.svelte:
```diff
@@ -0,0 +1,225 @@
+		display: flex;
+		flex-direction: column;
+		gap: 0.5rem;
+		input {
```
Inputs are already styled site-wide in base.scss, so you only need to edit them if you want them to be different from others. This stuff is mostly redundant and takes it out of theme, so I think we can just strike it.

miloridenour commented on src/routes/(authed)/TopHomeMenu.svelte:
```diff
@@ -49,6 +53,15 @@
+				withdrawSessionId = getTxSessionId();
+				toggleWithdraw(true);
+			},
+			icon: ArrowDown
```
Lets use BankNoteArrowDown for this and then change Deposit to use BankNoteArrowUp

miloridenour reviewed (changes requested):
Great start overall, a few things to change before its ready.

sag333ar committed:
addressed first feedback on PR - remove redundant css for input elements

sag333ar committed:
using BankNoteArrowDown & BankNoteArrowUp - as stated by milo

sag333ar committed:
removed bitcoin withdraw option as suggested by milo in the pull request

sag333ar committed:
handle evm account withdrawals

sag333ar committed:
pulled changes from main

sag333ar committed:
updated code to use the latest asset picker component

sag333ar committed:
showing max value when withdrawing

sag333ar committed:
addressed this pr feedback - https://github.com/vsc-eco/altera-app/pull/29\#discussion_r2535985981

sag333ar committed:
allowing hive-contact selection for evm accounts. https://github.com/vsc-eco/altera-app/pull/29\#discussion_r2535995389

sag333ar committed:
withdrawal is working

miloridenour commented on src/lib/sendswap/utils/sendOptions.ts:
```diff
@@ -295,6 +302,13 @@ const magiTransfer: TransferMethod = {
 	fees: 'No Fees'
 };
 
+const magiWithdraw: TransferMethod = {
```
This would be considered a magi transfer, and doesn't need its own type. This is part of the solving network constrains, and adding a different transaction type might mess this up.

miloridenour commented on src/lib/sendswap/stages/withdraw/HiveMainnetWithdraw.svelte:
```diff
@@ -0,0 +1,281 @@
+	<div class="sections">
+		<div class="section">
+			<span class="label-like">Hive Account</span>
+			<ContactSearchBox
```
This box deselects the person right away, it kind of flashes as selected and then goes back to blank

sag333ar committed:
addressed pull request feedback

miloridenour commented on src/lib/sendswap/stages/withdraw/HiveMainnetWithdraw.svelte:
```diff
@@ -0,0 +1,281 @@
+			<ArrowLeft size="32" />
+		</PillButton>
+	</div>
+	<SelectAssetFlattened
```
When this is open, you have to hide the back arrow from the parent so only the select back arrow shows. Should be an example in deposit.

sag333ar committed:
addressed following pull request comment
When this is open, you have to hide the back arrow from the parent so only the select back arrow shows. Should be an example in deposit.

sag333ar committed:
addressed following pull request comment
This is reference in its own scope. Easiest fix is to wrap this entire useMachine() call in a  rune.

sag333ar committed:
showing correct contacts for withdraw when user is using evm wallet

miloridenour committed:
removed contacts from withdrawl
